src/main/scala/gitbucket/core/service/RepositoryService.scala

```diff
@@ -434,7 +434,7 @@ trait RepositoryService { self: AccountService =>
     val preferred = PluginRegistry().renderableExtensions.map(extension => s"${fileBaseName}.${extension}")
 
     def choiceTemplate(files: List[FileInfo]): Option[FileInfo] =
-      files.find(f => preferred.contains(f.name)).orElse(files.find(f => f.name.startsWith(fileBaseName)))
+      files.find(f => preferred.contains(f.name)).orElse(files.find(f => fileBaseName == FileUtil.trimExtension(f.name)))
```

Comment:
can't we simplify with something like _(untested)_

`files.find(f => preferred == f.name).orElse(files.find => fileBaseName == f.name)`

with file extension removal we could select `ISSUE_TEMPLATE.TXT` instead of `ISSUE_TEMPLATE`
I would tend to KISS principle.